lint/pmd.bzl

```diff
@@ -83,11 +83,13 @@ def _pmd_aspect_impl(target, ctx):
 
     files_to_lint = filter_srcs(ctx.rule)
 
-    report, exit_code, info = report_files(_MNEMONIC, target, ctx)
+    outputs, info = output_files(_MNEMONIC, target, ctx)
     if len(files_to_lint) == 0:
-        dummy_successful_lint_action(ctx, report, exit_code)
+        dummy_successful_lint_action(ctx, outputs.human.stdout, outputs.human.exit_code)
     else:
-        pmd_action(ctx, ctx.executable._pmd, files_to_lint, ctx.files._rulesets, report, exit_code)
+        pmd_action(ctx, ctx.executable._pmd, files_to_lint, ctx.files._rulesets, outputs.human.stdout, outputs.human.exit_code)
+
+    pmd_action(ctx, ctx.executable._pmd, files_to_lint, ctx.files._rulesets, outputs.machine.stdout, outputs.machine.exit_code)
```

Comment:
same comment for pmd as for shellcheck, https://github.com/aspect-build/rules_lint/pull/331/files#r1680423092